cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,215 @@
+  private static final String DB_NAME = System.getenv("DB_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASSWORD = System.getenv("DB_PASS");
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+
+  private static class LazyInitializer {
+    private static final Tracer TRACER = Tracing.getTracer();
+    static {
+      try {
+        // Set up StackDriver
+        StackdriverTraceExporter.createAndRegister(
+            StackdriverTraceConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+      } catch (java.io.IOException e) {
+        Main.LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+      }
+
+      // Trace every request
+      TraceConfig traceConfig = Tracing.getTraceConfig();
+      traceConfig.updateActiveTraceParams(
+          traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample())
+              .build());
+    }
+
+    private LazyInitializer() {
+    }
+
+    private static Tracer getTracer() {
+      return LazyInitializer.TRACER;
+    }
+  }
+
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+    String connType = request.getFirstQueryParameter("conn_type").orElse("pool");
+
+    String requestId = UUID.randomUUID().toString();
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+    switch (connType) {
+      case "pool":
+        // Start tests for pooled connections
+        DataSource pool = createPool(jdbcURL, connProps);
+        execService.scheduleAtFixedRate(() -> {
+          connectWithPool(pool, requestId);
+        }, 0, interval, TimeUnit.MILLISECONDS);
+        break;
+      case "regular":
+        // Start tests for regular connections
+        execService.scheduleAtFixedRate(() -> {
+          connectRegular(jdbcURL, connProps, requestId);
+        }, 0, interval, TimeUnit.MILLISECONDS);
+        break;
+      default:
+        response.setStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+        BufferedWriter writer = response.getWriter();
+        writer.write("Valid options for conn_type are: pool, regular");
```

Comment:
nit: It's actually better to try to validate the request at the start of the function, so you return before doing any real work if it's invalid (not necessarily significant here since there isn't much work done before this point, but something to consider).

Reply:
(If not, make sure you return inside default instead of just breaking)

Reply:
fixed.